sqlmesh/core/context.py

```diff
@@ -2485,6 +2457,19 @@ def lint_models(
                 "Linter detected errors in the code. Please fix them before proceeding."
             )
 
+    def load_model_tests(
+        self, tests: t.Optional[t.List[str]] = None, patterns: list[str] | None = None
+    ) -> t.List[ModelTestMetadata]:
+        # If a set of tests is provided, use a single loader to load them
+        # Otherwise, gather all tests from all loaders/repos
+        loaders = [self._loaders[0]] if tests else self._loaders
+
+        model_tests = []
+        for loader in loaders:
+            model_tests.extend(loader._load_model_tests(tests=tests, patterns=patterns))
```

Comment:
Let's not rely on an internal method here, we don't do this elsewhere in context w.r.t. loading stuff. Perhaps you can follow the `load_materializations` approach?